### PR TITLE
Fixed a NameError on exception, added a unit test.

### LINKED ISSUE #130: NameError: name 'serviceName' is not defined
File "oauth2client/oauth2client/util.py", line 142, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "google-api-python-client/googleapiclient/discovery.py", line 193, in build
    content = _retrieve_discovery_doc(requested_url, http, cache_discovery, cache)
  File "googleapiclient/discovery.py", line 236, in _retrieve_discovery_doc
    raise UnknownApiNameOrVersion("name: %s  version: %s" % (serviceName,
NameError: name 'serviceName' is not defined

The same for `version` too, not passed to this function
Other http errors are also not handled.
I suggest you move out the codes inside function `_retrieve_discovery_doc`, and remove this function because it's only used once.
The code is messy

### PR DESCRIPTION
**Pull request by tmatsuo**
Fixes #130

### CODE REVIEW

**Comment by tmatsuo** [comment]
@nathanielmanistaatgoogle 

Sorry about not adding unit test for the exception clause.

File: googleapiclient/discovery.py [hunk 1]
```diff
@@ -190,7 +190,15 @@ def build(serviceName,
 
   requested_url = uritemplate.expand(discoveryServiceUrl, params)
 
-  content = _retrieve_discovery_doc(requested_url, http, cache_discovery, cache)
+  try:
+    content = _retrieve_discovery_doc(requested_url, http, cache_discovery,
+                                      cache)
+  except HttpError as e:
+    if e.resp.status == 404:
```
Comment (nathanielmanistaatgoogle) [inline googleapiclient/discovery.py]: httplib.NOT_FOUND rather than the literal value 404, please.

**Comment by nathanielmanistaatgoogle** [comment]
Thanks for the fix; just the one nitpick comment.

File: googleapiclient/discovery.py (hunk 1 above)
Comment (tmatsuo) [inline googleapiclient/discovery.py]: Done, httplib is only available with python2, so I needed a hack for py3 though.

**Comment by tmatsuo** [comment]
@nathanielmanistaatgoogle Thanks for the comment, PTAL

File: googleapiclient/discovery.py (hunk 1 above)
Comment (nathanielmanistaatgoogle) [inline googleapiclient/discovery.py]: Oof, no hack please. How about "from six.moves import http_client"? (https://pythonhosted.org/six/#module-six.moves)

Comment (tmatsuo) [inline googleapiclient/discovery.py]: I see, done. Thanks

**Comment by tmatsuo** [comment]
@nathanielmanistaatgoogle PTAL

**Comment by nathanielmanistaatgoogle** [comment]
Thanks!

**Pull request merged**